scipy/stats/_multicomp.py

```diff
@@ -24,10 +27,167 @@
 class DunnettResult:
     statistic: np.ndarray
     pvalue: np.ndarray
+    _alternative: Literal['two-sided', 'less', 'greater']
```

Comment:
These private attributes are showing up in `__repr__`. I'd suggest hiding them as in `EmpiricalDistributionFunction`'.

Reply:
mmm are you sure? If we say `__repr__` should allow to build back the object, then we would need (at least some of) the private attributes.

I don't have strong feelings on this one, I am not sure since we have a pretty `__str__`

Reply:
> If we say __repr__ should allow to build back the object

I think of that as the ideal distinction between `__repr__` and `__str__` if there is to be one. I would prefer to break that rule than to expose private attributes, though.

Reply:
They can't recreate the object anyway because it's private.